tests/Tests.AzureAppConfiguration/RefreshTests.cs

```diff
@@ -771,6 +771,37 @@ public async Task RefreshTests_UpdatesAllSettingsIfInitialLoadFails()
             Assert.Equal("TestValue3", configuration["TestKey3"]);
         }
 
+        [Fact]
+        public void RefreshTests_ResetCacheForcesNextRefresh()
+        {
+            IConfigurationRefresher refresher = null;
+            var mockClient = GetMockConfigurationClient();
+
+            var config = new ConfigurationBuilder()
+                .AddAzureAppConfiguration(options =>
+                {
+                    options.Client = mockClient.Object;
+                    options.Select("TestKey*");
+                    options.ConfigureRefresh(refreshOptions =>
+                    {
+                        refreshOptions.Register("TestKey1")
+                            .SetCacheExpiration(TimeSpan.FromDays(1));
+                    });
+
+                    refresher = options.GetRefresher();
+                })
+                .Build();
+
+            Assert.Equal("TestValue1", config["TestKey1"]);
+            FirstKeyValue.Value = "newValue";
+
+            // Invalidate the cache to force next refresh
+            refresher.ResetCache();
+
+            refresher.RefreshAsync().Wait();
+            Assert.Equal("newValue", config["TestKey1"]);
```

Comment:
Can we add these 2 lines before the `ResetCache()` too to validate values are not changed?

Reply:
Added